web-client/public/js/graph.js

```diff
@@ -485,11 +485,8 @@ export var drawGraph = function (network) {
         .attr("id", function (d) {
             return "path" + d.source.index + "_" + d.target.index;
         }).style("stroke-width", function (d) {
-            if (!grnState.colorOptimal) {
-                return d.strokeWidth = "2";
-            } else {
-                return d.strokeWidth = getEdgeThickness(d);
-            }
+            d.strokeWidth = grnState.colorOptimal ? getEdgeThickness(d) : 2;
+            return d.strokeWidth;
```

Comment:
This was the root of the problem. The code relies on the side effect of assigning the determined `strokeWidth` to the corresponding link object because this `strokeWidth` is accessed later in the code to determine marker choice, edge thickness, distance from a node, etc.—it turns out to affect a lot of other characteristics in the graph.

Although technically, an assignment statement in JavaScript does have a value (that being the right-side value which was assigned to the left side of the `=`), it is generally bad practice to treat an assignment as such. In this case, the assignment statement, when used this way, appears to "pick up" too late—i.e., the rest of the simulation seems to use the prior value of `strokeWidth` in rendering the graph for certain nodes.

Recoding this correctly—i.e., assigning the property first and then returning it—appears to eliminate that race condition. My local version no longer shows the anomalies and errors seen previously when the "Disable Formatting of Edges Based on Weight Value" is chosen. Hope you see the same improvement.

Reply:
understood, I completely overlooked this nuance of Javascript